pkg/sqlite/load.go

```diff
@@ -655,36 +694,47 @@ func (s *sqlLoader) RemovePackage(packageName string) error {
 			tx.Rollback()
 		}()
 
+		var errors []error
+		err = s.checkIfPackageExists(tx, packageName)
+		if err != nil {
+			errors = append(errors, err)
+		}
+
 		csvNames, err := s.getCSVNames(tx, packageName)
 		if err != nil {
-			return err
+			errors = append(errors, err)
 		}
 		for _, csvName := range csvNames {
 			if err := s.rmBundle(tx, csvName); err != nil {
-				return err
+				errors = append(errors, err)
 			}
 		}
 
 		deletePackage, err := tx.Prepare("DELETE FROM package WHERE package.name=?")
 		if err != nil {
-			return err
+			errors = append(errors, err)
 		}
 		defer deletePackage.Close()
 
 		if _, err := deletePackage.Exec(packageName); err != nil {
-			return err
+			errors = append(errors, err)
 		}
 
 		deleteChannel, err := tx.Prepare("DELETE FROM channel WHERE package_name = ?")
 		if err != nil {
-			return err
+			errors = append(errors, err)
 		}
 		defer deleteChannel.Close()
 
 		if _, err := deleteChannel.Exec(packageName); err != nil {
-			return err
+			errors = append(errors, err)
 		}
-		return tx.Commit()
+
+		if err = tx.Commit(); err != nil {
```

Comment:
Do we still want to commit the transaction if an error was returned from one of the statements? I am concerned that this will make it possible to perform a partial removal.